From 416e6f9033f55658561730a204ab44c647023b41 Mon Sep 17 00:00:00 2001
From: Murali Karicheri <m-karicheri2@ti.com>
Date: Tue, 16 Feb 2016 17:55:46 -0500
Subject: [PATCH 19/23] net: netcp: set proper ale revision mask for major
 version field

The mask for major version in ALE register is 4 bit instead of
8 bit as in other TI devices. So to support netcp, add a configurable
mask and use in the ALE driver.

Signed-off-by: Murali Karicheri <m-karicheri2@ti.com>
---
 drivers/net/ethernet/ti/netcp_ethss.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/ti/netcp_ethss.c b/drivers/net/ethernet/ti/netcp_ethss.c
index 7c72797..8d83a15 100644
--- a/drivers/net/ethernet/ti/netcp_ethss.c
+++ b/drivers/net/ethernet/ti/netcp_ethss.c
@@ -4263,6 +4263,8 @@ static int gbe_probe(struct netcp_device *netcp_device, struct device *dev,
 	ale_params.ale_ageout	= GBE_DEFAULT_ALE_AGEOUT;
 	ale_params.ale_entries	= gbe_dev->ale_entries;
 	ale_params.ale_ports	= gbe_dev->ale_ports;
+	if (IS_SS_ID_MU(gbe_dev))
+		ale_params.major_ver_mask = 0xf;
 
 	gbe_dev->ale = cpsw_ale_create(&ale_params);
 	if (!gbe_dev->ale) {
-- 
1.9.1

